tools/azsdk-cli/Azure.Sdk.Tools.Cli/Helpers/CodeownersValidatorHelper.cs

```diff
@@ -0,0 +1,162 @@
+
+                return result;
+            }
+            catch (NotFoundException)
+            {
+                result.Status = "Error";
+                result.Message = $"GitHub user '{username}' not found";
+                logger.LogWarning("GitHub user not found: {Username}", username);
+                return result;
+            }
+            catch (RateLimitExceededException ex)
+            {
+                result.Status = "Error";
+                result.Message = $"Rate limit exceeded. Reset at: {ex.Reset}";
+                logger.LogError("Rate limit exceeded for user {Username}. Reset at: {Reset}", 
+                    username, ex.Reset);
+                return result;
+            }
+            catch (SecondaryRateLimitExceededException ex)
+            {
+                result.Status = "Error";
+                result.Message = $"Secondary rate limit exceeded. Please wait a few minutes before trying again.";
+                logger.LogError("Secondary rate limit exceeded for user {Username}: {Message}", 
+                    username, ex.Message);
+                return result;
+            }
+            catch (Exception ex)
+            {
+                result.Status = "Error";
+                result.Message = $"Error validating user: {ex.Message}";
+                logger.LogError(ex, "Error validating GitHub user: {Username}", username);
+                return result;
+            }
+        }
+
+        /// <summary>
+        /// Validates the user's organization memberships (Microsoft and Azure only).
+        /// </summary>
+        private async Task<bool> ValidateOrganizationsAsync(string username, CodeownersValidationResult result)
+        {
+            try
+            {
+                var client = ((GitHubService)githubService).gitHubClient;
+                var organizations = await client.Organization.GetAllForUser(username);
+                var userOrgs = organizations.Select(org => org.Login).ToHashSet(StringComparer.OrdinalIgnoreCase);
+
+                // Check only required organizations
+                foreach (var requiredOrg in RequiredOrganizations)
+                {
+                    bool isMember = userOrgs.Contains(requiredOrg);
+                    result.Organizations[requiredOrg] = isMember;
+                }
+
+                return RequiredOrganizations.All(org => result.Organizations[org]);
+            }
+            catch (Exception ex)
+            {
+                logger.LogError(ex, "Error validating organizations for user: {Username}", username);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Validates the user's write permissions on azure-sdk-for-net repository.
+        /// </summary>
+        private async Task<bool> ValidatePermissionsAsync(string username)
+        {
+            try
+            {
+                var client = ((GitHubService)githubService).gitHubClient;
```

Comment:
Same comment from line 114. Don't leak `gitHubClient` into this class.

Instead, you could do something on `IGitHubService` like: 

```
public Task<bool> HasWritePermission(string org, string repo, string username)
```

Also, this only checks the user's permission against a single repo. It'd be better if we checked the user's permission against the current repo (e.g. specs if they're in specs, .NET if they're in .NET, etc.)